labels/labels.go

```diff
@@ -103,6 +103,23 @@ func (ls Labels) Map() map[string]string {
 	return m
 }
 
+// WithoutEmpty returns the labelset without empty labels.
+// May return the same labelset.
+func (ls Labels) WithoutEmpty() Labels {
```

Comment:
Maybe me personal opinion, but I would say just function is better here as it has to be done only once at certain code logic.

Having this as Labels method indicates you need to do it a lot... but we don't want unnecceary run through all labels if not needed.

Reply:
If this turns out to be a bigger problem than I think, we'll need to sprinkle this across the codebase.

Reply:
I don't feel strongly either way.